src/frontend/src/scheduler/distributed/stage.rs

```diff
@@ -384,27 +366,22 @@ impl StageRunner {
                                     // All tasks in this stage have been scheduled. Notify query runner to schedule next
                                     // stage.
                                     if running_task_cnt == self.tasks.keys().len() {
-                                        self.notify_schedule_next_stage().await?;
+                                        self.notify_schedule_next_stage().await;
                                         sent_signal_to_next = true;
                                     }
                                 }
 
                                 TaskStatusProst::Failed => {
-                                    // If receive task failure, report to query runner and abort tasks.
-                                    let task_execution_err = SchedulerError::TaskExecutionError;
-                                    self.send_event(QueryMessage::Stage(StageEvent::Failed {id: self.stage.id, reason: task_execution_err})).await?;
-                                    self.abort_all_running_tasks().await?;
-
-                                    break;
+                                    // Throw the error and caller write the event to channel.
+                                    return Err(SchedulerError::TaskExecutionError);
                                 }
 
-                                TaskStatusProst::Finished | TaskStatusProst::Aborted => {
+                                TaskStatusProst::Finished => {
```

Comment:
If all task finished, it should change stage state to finished, and when query finished, it should be sent to some cleanup queue to be cleaned, otherwise we will accumulate queries forever.

Reply:
Yes. I plan to do this later. After some refactor in #4924